system/stackdepth.c

```diff
@@ -0,0 +1,44 @@
+/* stackdepth.c - stackdepth */
+
+/* Group Members: Seth Redwine, Nick Palacio, Jeffrey Allen */
+
+#include <xinu.h>
+
+/*------------------------------------------------------------------------
+ * stackdepth - determines how many nested function calls have been made
+ * by the current process
+ *------------------------------------------------------------------------
+ */
+
+int stackdepth() {
+
+	struct procent *proc = &proctab[currpid];
+
+	unsigned long *top_esp;
+	unsigned long *top_ebp;
+
+	unsigned long *sp;
+	unsigned long *fp;
+
+    int count = 0;
+
+	asm("movl %%esp, %0" : "=r" (top_esp));
```

Comment:
What do these lines do?

Reply:
Those lines read the values of the "esp" and "ebp" registers and save them into local variables. I used the inline assembly language function as specified by the assignment instructions:

"To access the last stack frame pushed onto the run‐time stack, use the
in‐line assembly function asm() to read the values of %esp and %ebp and save them
into local variables".

Reply:
I believe him